ui/components/app/network-display/network-display.js

```diff
@@ -1,129 +1,27 @@
- * Please update your code to use the new `<PickerNetwork>` component instead, which can be found at ui/components/component-library/picker-network/picker-network.tsx.
- * You can find documentation for the new `PickerNetwork` component in the MetaMask Storybook:
- * {@link https://metamask.github.io/metamask-storybook/?path=/docs/components-componentlibrary-pickernetwork--docs}
- * If you would like to help with the replacement of the old `NetworkDisplay` component, please submit a pull request against this GitHub issue:
- * {@link https://github.com/MetaMask/metamask-extension/issues/20485}
- */
-
-export default function NetworkDisplay({
-  indicatorSize,
-  disabled,
-  labelProps,
-  targetNetwork,
-  onClick,
-}) {
-  const networkIsLoading = useSelector(isNetworkLoading);
-  const providerConfig = useSelector(getProviderConfig);
-  const t = useI18nContext();
-
-  const { nickname, type: networkType } = targetNetwork ?? providerConfig;
+import { getCurrentNetwork } from '../../../selectors';
+import { PickerNetwork, AvatarNetworkSize } from '../../component-library';
 
+export default function NetworkDisplay() {
+  const currentNetwork = useSelector(getCurrentNetwork);
   return (
-    <Chip
-      dataTestId="network-display"
-      borderColor={
-        onClick ? BorderColor.borderDefault : BorderColor.borderMuted
-      }
-      onClick={onClick}
-      leftIcon={
-        <LoadingIndicator
-          alt={t('attemptingConnect')}
-          title={t('attemptingConnect')}
-          isLoading={networkIsLoading}
-        >
-          <ColorIndicator
-            color={
-              networkType === NETWORK_TYPES.RPC
-                ? IconColor.iconMuted
-                : networkType
-            }
-            size={indicatorSize}
-            type={ColorIndicator.TYPES.FILLED}
-            iconClassName={
-              networkType === NETWORK_TYPES.RPC && indicatorSize !== Size.XS
-                ? 'fa fa-question'
-                : undefined
-            }
-          />
-        </LoadingIndicator>
-      }
-      rightIcon={
-        onClick ? <Icon name={IconName.ArrowDown} size={IconSize.Xs} /> : null
-      }
-      label={
-        networkType === NETWORK_TYPES.RPC
-          ? nickname ?? t('privateNetwork')
-          : t(getNetworkLabelKey(networkType))
-      }
-      className={classnames('network-display', {
-        'network-display--disabled': disabled,
-        'network-display--clickable': typeof onClick === 'function',
-      })}
-      labelProps={{
-        variant: TypographyVariant.H7,
-        ...labelProps,
-      }}
+    <PickerNetwork
+      className="network-display"
+      label={currentNetwork?.nickname}
+      labelProps={{ 'data-testid': 'network-display' }}
+      src={currentNetwork?.rpcPrefs?.imageUrl}
+      iconProps={{ display: 'none' }} // do not show the dropdown icon
```

Comment:
Thoughts on making the icon smaller? It seems to be too close to the container's borders

Reply:
Agree we need some design attention here. The minimum size is `AvatarNetworkSize.Xs` which is what is being used

Reply:
Looks like this has been updated

<img width="661" alt="Screenshot 2024-06-26 at 2 20 18 PM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/e8083e4e-d8e7-4503-aaad-aad0d10132d8">